app/controllers/export_file_controller.rb

```diff
@@ -22,21 +22,47 @@ def start
     @id = params[:id]
   end
 
-  def export
-    @delim_type = params[:delim_type]
-    if @delim_type == "comma"
-      filename = params[:model] + params[:id] + ".csv"
+  # Find the filename and delimiter
+  def find_delim_filename(delim_type, other_char, suffix = "")
+    if delim_type == "comma"
+      filename = params[:model] + params[:id] + suffix + ".csv"
       delimiter = ","
-    elsif @delim_type == "space"
-      filename = params[:model] + params[:id] + ".csv"
+    elsif delim_type == "space"
+      filename = params[:model] + params[:id] + suffix + ".csv"
       delimiter = " "
-    elsif @delim_type == "tab"
-      filename = params[:model] + params[:id] + ".csv"
+    elsif delim_type == "tab"
+      filename = params[:model] + params[:id] + suffix + ".csv"
       delimiter = "\t"
-    elsif @delim_type == "other"
-      filename = params[:model] + params[:id] + ".csv"
+    elsif delim_type == "other"
+      filename = params[:model] + params[:id] + suffix + ".csv"
       delimiter = other_char
     end
+    return filename, delimiter
```

Comment:
The code climate complains about the redundant return statement but while returning two values the return statement is mandatory, else the controller throws an error.